src/binder/clrprivbindercoreclr.cpp

```diff
@@ -157,10 +149,12 @@ HRESULT CLRPrivBinderCoreCLR::BindUsingPEImage( /* in */ PEImage *pPEImage,
                 hr = BindAssemblyByNameWorker(pAssemblyName, &pCoreCLRFoundAssembly, true /* excludeAppPaths */);
                 if (SUCCEEDED(hr))
                 {
+                    // Is identity same as TPA assembly
                     if (pCoreCLRFoundAssembly->GetIsInGAC())
                     {
-                        // If we were able to bind to a TPA assembly, then fail the load
-                        IF_FAIL_GO(HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND));
+                        // Fail bind if requested assembly is at different location than TPA assembly
+                        if(pPEImage != pCoreCLRFoundAssembly->GetPEImage())
```

Comment:
Yes if spec does not match what is present in TPA then loadfrom will fail for default context. I have verified this.